### DISCUSSION TITLE
A way to dedupe `Alpine.data()` registrations?

### DISCUSSION THREAD

**Discussion by DEV-ONI** [alpinejs/alpine#3062] [1. Help]
Is there a way to dedupe `Alpine.data` component registration? I could not find any docs on this behavior.

Say I want to register a component -

```js
// first time 
Alpine.data('someComponent' () => {...})


// once again(second time)
Alpine.data('someComponent' () => {...})
```

Is there a way to prevent duplicate registrations of the same component?
Or what is the behavior in this case?

**Comment by SimoTod** [comment]
the second call will override the first one.

**Reply by DEV-ONI** [reply]
How can I prevent it from overriding the component?
Is there an API-exposed method to check against all registered components?

**Reply by SimoTod** [reply]
Nothing I'm aware of.

**Comment by benkoshy** [comment]
> Is there a way to dedupe `Alpine.data` component registration? I could not find any docs on this behavior.
> 
> Say I want to register a component -
> 
> ```js
> // first time 
> Alpine.data('someComponent' () => {...})
> 
> 
> // once again(second time)
> Alpine.data('someComponent' () => {...})
> ```
> 
> Is there a way to prevent duplicate registrations of the same component? Or what is the behavior in this case?

Can you share your use case? To consider other solutions
